test/Microsoft.VisualStudio.LanguageServices.Razor.Test/DefaultTagHelperCompletionFactsServiceTest.cs

```diff
@@ -0,0 +1,407 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Xunit;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    public class DefaultTagHelperCompletionFactsServiceTest
+    {
+        [Fact]
+        public void GetElementCompletions_AllowsMultiTargetingTagHelpers()
+        {
+            // Arrange
+            var documentDescriptors = new[]
+            {
+                ITagHelperDescriptorBuilder.Create("BoldTagHelper1", "TestAssembly")
+                    .TagMatchingRule(rule => rule.RequireTagName("strong"))
+                    .TagMatchingRule(rule => rule.RequireTagName("b"))
+                    .TagMatchingRule(rule => rule.RequireTagName("bold"))
+                    .Build(),
+                ITagHelperDescriptorBuilder.Create("BoldTagHelper2", "TestAssembly")
+                    .TagMatchingRule(rule => rule.RequireTagName("strong"))
+                    .Build(),
+            };
+            var documentContext = TagHelperDocumentContext.Create(string.Empty, documentDescriptors);
+            var service = CreateTagHelperCompletionFactsService();
+            var expectedCompletions = ElementCompletionResult.Create(new Dictionary<string, IEnumerable<TagHelperDescriptor>>()
+            {
+                ["strong"] = new[] { documentDescriptors[0], documentDescriptors[1] },
+                ["b"] = new[] { documentDescriptors[0] },
+                ["bold"] = new[] { documentDescriptors[0] },
+            });
```

Comment:
It would be really helpful if you could pull this guy out with some whitespace to make it easier to read. All of this code needs to breathe.